src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -302,7 +302,7 @@ private void RemoveInstrumentState(Instrument instrument, InstrumentState state)
                 {
                     lock (this)
                     {
-                        return (!CheckTimeSeriesAllowed() || !CheckHistogramAllowed()) ?
```

Comment:
Could you please add some comment here telling the check order matter?

Reply:
I added comment about fix.Could you please review it?